api/src/handler.js

```diff
@@ -27,61 +27,54 @@ export default harden(({ board, publicFacet, invitationIssuer }, _invitationMake
+          } = data;
+
+          // A dehydrated amount has the form: { brand:
+          // brandBoardId, value }
 
-              // A dehydrated amount has the form: { brand:
-              // brandBoardId, value }
+          // dehydratedBrandOut is a brandBoardId
+          const [amountIn, brandOut] = await Promise.all([
+            hydrateAmount(dehydratedAmountIn),
+            hydrateBrand(dehydratedBrandOut)
+          ]);
+          const { value } = await E(publicFacet).getInputPrice(amountIn, brandOut);
+          send({ type: 'autoswap/getInputPriceResponse', data: value });
+          return true;
+        }
 
-              // dehydratedBrandOut is a brandBoardId
-              const [amountIn, brandOut] = await Promise.all([
-                hydrateAmount(dehydratedAmountIn), 
-                hydrateBrand(dehydratedBrandOut)
-              ]);
-              const { value } = await E(publicFacet).getInputPrice(amountIn, brandOut);
-              return { type: 'autoswap/getInputPriceResponse', data: value };
-            }
+        case 'autoswap/sendSwapInvitation': {
+          const { depositFacetId, offer } = obj.data;
+          const depositFacet = E(board).getValue(depositFacetId);
+          const invitation = await E(publicFacet).makeSwapInvitation();
+          const invitationAmount = await E(invitationIssuer).getAmountOf(invitation);
+          const { value: [{ handle }] } = invitationAmount;
+          const invitationHandleBoardId = await E(board).getId(handle);
+          const updatedOffer = { ...offer, invitationHandleBoardId };
+          // We need to wait for the invitation to be
+          // received, or we will possibly win the race of
+          // proposing the offer before the invitation is ready.
+          // TODO: We should make this process more robust.
+          await E(depositFacet).receive(invitation);
 
-            case 'autoswap/sendSwapInvitation': {
-              const { depositFacetId, offer } = obj.data;
-              const depositFacet = E(board).getValue(depositFacetId);
-              const invitation = await E(publicFacet).makeSwapInvitation();
-              const invitationAmount = await E(invitationIssuer).getAmountOf(invitation);
-              const { value: [{ handle }]} = invitationAmount;
-              const invitationHandleBoardId = await E(board).getId(handle);
-              const updatedOffer = { ...offer, invitationHandleBoardId };
-              // We need to wait for the invitation to be
-              // received, or we will possibly win the race of
-              // proposing the offer before the invitation is ready.
-              // TODO: We should make this process more robust.
-              await E(depositFacet).receive(invitation);
-              
-              return harden({
-                type: 'autoswap/sendSwapInvitationResponse',
-                data: { offer: updatedOffer },
-              });
-            }
+          send({
```

Comment:
shouldn't this be waiting for the invitation to be delivered?  Otherwise there's a race to tell the wallet to make an offer against an invitation that has not yet arrived.

Reply:
Line 65 is doing this await. I hope to remove the need for this soon, though!